quarkus/tests/integration/src/test/java/org/keycloak/it/storage/database/dist/DatasourcesDistTest.java

```diff
@@ -0,0 +1,22 @@
+package org.keycloak.it.storage.database.dist;
+
+import io.quarkus.test.junit.main.Launch;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.keycloak.it.junit5.extension.CLIResult;
+import org.keycloak.it.junit5.extension.DistributionTest;
+import org.keycloak.it.junit5.extension.RawDistOnly;
+import picocli.CommandLine;
+
+@DistributionTest
+@RawDistOnly(reason = "Containers are immutable")
+@Tag(DistributionTest.STORAGE)
+public class DatasourcesDistTest {
+
+    @Test
+    @Launch({"start-dev", "--db-multiple-datasources=true", "--db-kind-users=postgres", "--db-kind-clients=postgres"})
+    public void multipleNonXaDatasources(CLIResult result) {
+        // TODO error: Build time option: '--db-kind-users' not usable with build
```

Comment:
@shawkins Maybe you will know more about these wildcards.

Issue: 
When I run:
```
start-dev --db-kind-users=postgres
```
I got an error: 
```
Build time option: '--db-kind-users' not usable with build
```

- The reason is that when running reaugmentation, Picocli does not know('Unknown option: xx') the `db-kind-users` property, only `db-kind-<datasource>`
- Option `db-kind-<datasource>` is build time.
- Everything works when I run first `build` and then `start` with `--optimized`
- Until now, we had only runtime wildcard, and I think we have some gaps for supporting build time wildcard mappers. 

@shawkins Do you have anything in mind that would need to be updated?

Reply:
> Do you have anything in mind that would need to be updated?

It should be some kind of problem with unrecognizedArgs check being performed by validateConfig. It should be an allowed mapper.

Reply:
@shawkins It does not even get into the `Picocli.validateConfig()`.

The exception is thrown directly when executing Picocli cmd: https://github.com/keycloak/keycloak/blob/main/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java#L309

But weird thing is that it works for `build`, but not when reaug is executed. I'm trying to find some differences between them in terms of Picocli args or mapping.

Reply:
Were you able to get past this? We could catch up tomorrow in my morning if that works for you.

Reply:
@shawkins I'll update other stuff and probably focus on this next week, but thanks!

Reply:
I disabled the immediate fail for unmatched arguments for the `./kc.sh` cmd. All these validations are done later. I hope it's ok.

https://github.com/keycloak/keycloak/pull/37640/files#diff-5e7526010bd1f3d3c2735d6b5d600aaf3457a83eb1699ef7e10c91e2257ae6d2R314